pkg/iac/scanners/terraform/executor/executor.go

```diff
@@ -130,10 +130,16 @@ func writeBlock(tfBlock *terraform.Block, block *hclwrite.Block, causeRng types.
 	var found bool
 
 	for _, attr := range tfBlock.Attributes() {
-		if attr.GetMetadata().Range().Covers(causeRng) && !attr.IsLiteral() {
-			block.Body().SetAttributeValue(attr.Name(), attr.Value())
-			found = true
+		if !attr.GetMetadata().Range().Covers(causeRng) || attr.IsLiteral() {
+			continue
+		}
+
+		value := attr.Value()
+		if !value.IsKnown() || value.IsNull() {
+			continue
```

Comment:
Should we also check if it is `cty.NilVal`?
```suggestion
		if value == cty.NilVal || !value.IsKnown() || value.IsNull() {
			continue
```

Reply:
`IsNull` method will return true for `cty.NilVal`, so this is not necessary.

Reply:
What happens if you call `value.IsKnown()` on `cty.NilVal`? I realized we don't have a test for `writeBlock()` as `TestConvertObjectWithUnknownAndNullValuesToMap` doesn't cover it today does it?

Reply:
`IsKnown` checks that the value is not of type `unknownType`. It will return false for `cty.NilVal`.

Reply:
There is a [ complex test](https://github.com/aquasecurity/trivy/blob/fe400ea55fce444dec0579309a00c8da33c402b7/pkg/iac/scanners/terraform/scanner_test.go#L998) that covers it.